framework/include/base/MooseApp.h

```diff
@@ -544,6 +544,11 @@ class MooseApp : public ConsoleStreamInterface, public libMesh::ParallelObject
    */
   const RestartableDataMap & getRestartableDataMap(const RestartableDataMapName & name) const;
 
+  /**
+   * Return a writable reference to restartable data for the specific type flag.
+   */
+  RestartableDataMap & setRestartableDataMap(const RestartableDataMapName & name);
```

Comment:
Returning writeable references breaks encapsulation. I know we have it in other places throughout MOOSE (and even libMesh), but I want to avoid it as much as possible. Would it be possible to make this `void` and add another argument to this function instead?